code/obj/machinery/espresso.dm

```diff
@@ -180,24 +197,44 @@
 			src.add_fingerprint(user)
 			if (src.my_carafe) //freaking spacing errors made me waste hours on this
 				if (!(status & (NOPOWER|BROKEN)))
-					switch (alert("What would you like to do with [src]?",,"Brew coffee","Remove carafe","Nothing"))
-						if ("Brew coffee")
-							for(var/obj/item/reagent_containers/food/drinks/carafe/C in src.contents)
-								C.reagents.add_reagent("coffee_fresh",100)
-								playsound(src.loc, 'sound/misc/pourdrink.ogg', 50, 1)
-						if ("Remove carafe")
-							if (!src.my_carafe)
-								user.show_text("The carafe is gone!")
+					if(!src.emagged)
+						switch (alert("What would you like to do with [src]?",,"Brew coffee","Remove carafe","Nothing"))
+							if ("Brew coffee")
+								for(var/obj/item/reagent_containers/food/drinks/carafe/C in src.contents)
+									C.reagents.add_reagent("coffee_fresh",100)
+									playsound(src.loc, 'sound/misc/pourdrink.ogg', 50, 1)
```

Comment:
un-indent because if statement was removed, and make the "brew coffee" action also brew tea when emagged
```suggestion
						if ("Brew coffee","Brew tea")
							for(var/obj/item/reagent_containers/food/drinks/carafe/C in src.contents)
								C.reagents.add_reagent(src.emagged ? "tea" : "coffee_fresh",100)
								playsound(src.loc, 'sound/misc/pourdrink.ogg', 50, 1)
```